src/services.py

```diff
@@ -449,3 +454,152 @@ def ban_user(self, my_user_id, other_user_id):
                     self.BannedPair(buyer_id=other_user_id, seller_id=my_user_id),
                 ]
             )
+
+
+def chat_serializer(chat_room_result, chat_result, buyer, seller, user_id):
+    (author, dealer) = (
+        (seller, buyer) if seller.get("id") == user_id else (buyer, seller)
+    )
+    return {
+        "dealer_name": dealer.get("full_name"),
+        "dealer_id": dealer.get("id"),
+        "created_at": datetime.timestamp(chat_result.get("created_at")),
+        "updated_at": datetime.timestamp(chat_result.get("updated_at")),
+        "author_name": author.get("full_name"),
+        "author_id": author.get("id"),
+        "message": chat_result.get("message"),
+        "chatRoom_id": chat_room_result.get("id"),
+    }
+
+
+class ChatService(DefaultService):
+    def __init__(
+        self, config, User=User, UserService=UserService, Chat=Chat, ChatRoom=ChatRoom
+    ):
+        self.Buyer = aliased(User)
+        self.Seller = aliased(User)
+        self.User = User
+        self.Chat = Chat
+        self.UserService = UserService
+        self.ChatRoom = ChatRoom
+        self.config = config
+
+    def add_message(self, chat_room_id, message, author_id):
+        with session_scope() as session:
+            chat = Chat(
+                chat_room_id=str(chat_room_id),
+                message=message,
+                author_id=str(author_id),
+            )
+            session.add(chat)
+            session.flush()
+            session.refresh(chat)
+            chat = chat.asdict()
+
+            result = (
+                session.query(self.ChatRoom, self.Buyer, self.Seller)
+                .filter_by(id=chat.get("chat_room_id"))
+                .outerjoin(self.Buyer, self.Buyer.id == self.ChatRoom.buyer_id)
+                .outerjoin(self.Seller, self.Seller.id == self.ChatRoom.seller_id)
+                .all()[0]
```

Comment:
Is this only supposed to return one value only? Try `.one()` or `.one_or_none()`